packages/headless/src/features/facets/facet-set/facet-set-analytics-actions-utils.ts

```diff
@@ -94,20 +94,22 @@ export const buildFacetStateMetadata = (
     );
 
     if (isCategoryFacetRequest(facetRequest)) {
-      const hasSelectedValue = !!categoryFacetRequestSelectedValuesSelector(
+      const activeValues = categoryFacetRequestActiveValuesSelector(
         state,
         facetRequest.facetId
-      ).length;
+      );
 
-      if (!hasSelectedValue) {
+      if (!activeValues.length) {
         return;
       }
 
+      const lowestLeaf = activeValues[activeValues.length - 1];
```

Comment:
Not sure if this is the right way to handle this. I try to grab the absolute lowest level of the "tree" in case one of the parent values is selected, but a child value is excluded. Not sure if this even a possibility to be honest, but I wasn't sure so I assumed it's best to try and handle it.

Curious to hear what everyone thinks on this.